lib/errors.js

```diff
@@ -28,140 +28,135 @@ var BaseError = Base.inherits(Error, {
             Error.captureStackTrace(this, this.constructor);
         }
 
-        // Template the message given the info object
-        this.message = this.template(info);
-
-        // copy the code from the prototype to the instance so it will
-        // be serialized.
-        this.code = this.code;
-
+        this.code = code;
+        this.status = status;
         this.info = info;
     }
-});
+}
 
 var errors = {};
 
 // An error that can not be identified. Ideally, this error should
 // never be thrown. It is only used as a catchall in cases where a
 // more specific error can not be found.
-errors.UnknownError = BaseError.extend({
-    name: 'UnknownError',
-    code: 'JS-UNKNOWN-ERROR',
-    status: 500
-});
+class UnknownError extends BaseError {
+    constructor(info) {
+        super('UnknownError', 'JS-UNKNOWN-ERROR', 500, info);
+    }
+}
 
 
-// All errors from the juttle compiler and runtime are grouped as this
+// All var from the juttle compiler and runtime are grouped as this
 // error. The info.err field contains the complete error object from
 // the Juttle compiler or runtime, and the info.bundle field should
 // contain a program bundle containing the program and modules that
 // resulted in the error.
 
-errors.JuttleError = BaseError.extend({
-    name: 'JuttleError',
-    code: 'JS-JUTTLE-ERROR',
-    status: 400
-});
-
-errors.BundleError = BaseError.extend({
-    name: 'BundleError',
-    code: 'JS-BUNDLE-ERROR',
-    status: 400
-});
-
-errors.JobNotFoundError = BaseError.extend({
-    name: 'JobNotFoundError',
-    code: 'JS-JOB-NOT-FOUND-ERROR',
-    status: 404
-});
-
-errors.FileNotFoundError = BaseError.extend({
-    name: 'FileNotFoundError',
-    code: 'JS-FILE-NOT-FOUND-ERROR',
-    status: 404
-});
-
-errors.FileAccessError = BaseError.extend({
-    name: 'FileAccessError',
-    code: 'JS-FILE-ACCESS-ERROR',
-    status: 403
-});
-
-errors.DirectoryNotFoundError = BaseError.extend({
-    name: 'DirectoryNotFoundError',
-    code: 'JS-DIR-NOT-FOUND-ERROR',
-    status: 404
-});
-
-errors.DirectoryAccessError = BaseError.extend({
-    name: 'DirectoryAccessError',
-    code: 'JS-DIR-ACCESS-ERROR',
-    status: 403
-});
-
-errors.InvalidPathError = BaseError.extend({
-    name: 'InvalidPathError',
-    code: 'JS-INVALID-PATH-ERROR',
-    status: 400
-});
-
-errors.TimeoutError = BaseError.extend({
-    name: 'TimeoutError',
-    code: 'JS-TIMEOUT-ERROR',
-    status: 408
-});
+class JuttleError extends BaseError {
```

Comment:
Yeah I know... As it stands the conventions are a little different.

I'd be ok with adding the classes to exports if we need them.